spring-security-cas-extension/src/main/java/com/kakawait/spring/security/cas/client/CasClientProperties.java

```diff
@@ -0,0 +1,33 @@
+package com.kakawait.spring.security.cas.client;
+
+import java.util.Objects;
+
+/**
+ * @author Jonathan Coueraud
+ */
+public class CasClientProperties {
+
+    private String proxyTicketQueryKey = "ticket";

```

Comment:
You must use `getArtifactParameter()` from `ServiceProperties` https://github.com/spring-projects/spring-security/blob/master/cas/src/main/java/org/springframework/security/cas/ServiceProperties.java#L99

Each it should be an existing bean https://github.com/kakawait/cas-security-spring-boot-starter/blob/master/cas-security-spring-boot-autoconfigure/src/main/java/com/kakawait/spring/boot/security/cas/CasSecurityAutoConfiguration.java#L65

Reply:
Thus I think this class could be removed